lib/figures/pawn.rb

```diff
@@ -0,0 +1,63 @@
+# frozen_string_literal: true
+
+require_relative 'figure'
+
+class Pawn
+  attr_reader :row, :column, :possible_moves, :color, :display
+
+  include Figure
+
+  def initialize(color, row, column)
+    @color = color
+    @row = row
+    @column = column
+    @display = @display = { white: '♟', black: '♙' }
+    @move_pattern = find_move_pattern
+    @possible_moves = find_possible_moves
+  end
+
+  def find_move_pattern
+    result = []
+    if @color == :white
+      result.push([-2, 0]) if @row == 6
+      result.push([-1, 0])
+    else
+      result.push([2, 0]) if @row == 1
+      result.push([1, 0])
+    end
+  end
+
+  def find_possible_moves
+    @move_pattern = find_move_pattern
+    result = []
+    @move_pattern.each do |option|
+      row = @row + option[0]
+      column = @column + option[1]
+      if (row in 0..7) && (column in 0..7)
+        result.push([row, column])
+      end
+    end
+    result
+  end
+
+  def update_possible_moves(board)
```

Comment:
This is the only instance, where the moves depend on the situation on the board. I think, that this case invalidates the architecture, in which it's the responsibility of pieces to know where they can go, because now you have circular dependency: board depends on the pieces, and pieces depend on the board.

Circular dependencies are evil and should be avoided at all costs, I would advice you think about a different architecture: one where pieces just store theoretical possible move patterns, and board is responsible for finding out where a piece can actually go.

I know this might seem insignificant, but in actual rails apps this is exactly the kind of thing that leads to major architectural issues several months/years down the line.